spinalcordtoolbox/scripts/sct_dmri_moco.py

```diff
@@ -203,8 +217,23 @@ def main(argv: Sequence[str]):
     if not (is_qc_none == is_seg_none):
         parser.error("Both '-qc' and '-qc-seg' are required in order to generate a QC report.")
 
-    # run moco
-    fname_output_image = moco_wrapper(param)
+    # Run moco
+    if arguments.mocodl:
+        if run_mocoDL is None:
+            raise ImportError("mocoDL module not found. Please ensure SCT was installed with mocoDL support.")
+
+        fname_output_image = run_mocoDL(
+            fname_data=param.fname_data,
+            fname_mask=param.fname_mask,
+            ofolder=param.path_out,
+            fname_ref=param.fname_ref,
+            fname_bvecs=param.fname_bvecs,
+            fname_bvals=param.fname_bvals,
+            mode="dmri"
+        )
+    else:
+        # Run SCT-based motion correction
+        fname_output_image = moco_wrapper(param)
```

Comment:
I see that the new approach basically ignores the advanced `-param` settings made for `moco_wrapper` (e.g. poly, smooth, metric, etc.) as well as some other arguments (`-g`, `-bvalmin`, `-x`).

I think we need to be careful here with our interfaces, since users might mistakenly supply unsupported arguments.

Reply:
I think there are a few different options we could consider:

1. Simply checking for unsupported arguments and throwing an error if they are provided with `-dl`.
2. Adopting a more complex interface (e.g. subparsers) allowing for a different set of arguments to be displayed if the user selects `dl`.

I think I prefer the first simpler option? Just because rewriting the interface would be backwards incompatible, which is especially critical for a tool like moco which has seen quite a lot of use.